### PR TITLE
Support generating a README from a Plugin

### PR DESCRIPTION
**Pull request by orta**
Output - https://github.com/danger/danger/blob/lint/dsl_docs.md

Abstracts the plugin path logic from `danger plugin lint` into a new class, and adds `dance plugin readme` which can generate a markdown README chunk for a plugin. The doc above is generated by running:

```sh
bundle exec danger plugin readme lib/danger/danger_core/plugins/*.rb
```

This PR is mergable after review, but the work to make the DSL docs a nicer read is ongoing, I must have wasted 2-3 hours now on trying to make a table [like this](https://github.com/ReactiveX/RxSwift#usage) work, but restricting all code samples to a tiny amount of chars didn't fly in the end.

Need to think a bit about how to display the methods overall, but I'm happy with where it is ATM.

Ideally, this can be the foundation for our documentation ( as seen in https://github.com/danger/danger/issues/280 ) in the README, and on the site. Perhaps we could do something interesting with the summary thing:

<details><summary>This thing.</summary><code style="white-space:nowrap;">Right? Could add more inline docs here...</code></details>

### CODE REVIEW

File: lib/danger/commands/plugins/plugin_abstract.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+module Danger
+  class PluginAbstract < Runner
+    require 'danger/commands/plugins/plugin_lint'
+	require 'danger/commands/plugins/plugin_readme'
```
Comment (segiddins) [inline lib/danger/commands/plugins/plugin_abstract.rb]: alignment

File: lib/danger/danger_core/dangerfile.rb [hunk 2]
```diff
@@ -119,6 +119,9 @@ def method_values_for_plugin_hashes(plugin_hashes)
           # So that we either have one value per row
           # or we have [] for an empty array
           value = value.join("\n") if value.kind_of?(Array) && value.count > 0
```
Comment (segiddins) [inline lib/danger/danger_core/dangerfile.rb]: might read cleaner as a switch statement?

File: lib/danger/plugin_support/plugin_file_resolver.rb [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+          # and uses the latest version from the server
+          bundler = Bundler::Definition.new(nil, deps, source, true)
+          bundler.resolve_remotely!
+
+          # Install the gems into a tmp dir
+          options = { path: dir }
+          Bundler::Installer.install(Pathname.new(dir), bundler, options)
+
+          # Get the name'd gems out of bundler, then pull out all their paths
+          gems = gem_names.map { |name| bundler.specs[name] }.flatten
```
Comment (segiddins) [inline lib/danger/plugin_support/plugin_file_resolver.rb]: why not `flat_map` here & on the next line?

File: lib/danger/danger_core/dangerfile.rb (hunk 2 above)
Comment (orta) [inline lib/danger/danger_core/dangerfile.rb]: yeah, might express the intent better too

**Comment by orta** [comment]
Thanks @segiddins 👍

**Pull request merged**